app/routes.py

```diff
@@ -1,2 +1,257 @@
+        db.session.commit()
+        return {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+            }
+        }
+
+
+# G O A L S   E N D P O I N T S
+
+
+@goals_bp.route("", methods=["POST", "GET"])
+def handle_goals():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        new_goal = Goal(title=request_body["title"])
+        db.session.add(new_goal)
+        db.session.commit()
+        new_goal_response = {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title,
+            }
+        }
+        return jsonify(new_goal_response), 201
+
+    elif request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = []
+        for goal in goals:
+            goals_response.append(
+                {
+                    "id": goal.goal_id,
+                    "title": goal.title
+                }
+            )
+        return jsonify(goals_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        return {
+            "goal": {
+                "id": goal.goal_id,
+                "title": goal.title
+            }
+        }
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+        db.session.commit()
+        return {
+            "goal": {
+                "id": goal.goal_id,
+                "title": goal.title
+            }
+        }
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        return {
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }
+
+# G O A L S / T A S K S   E N D P O I N T S
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST", "GET"])
+def handle_goals_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "POST":
+        request_body = request.get_json()
+        task_ids = request_body["task_ids"]
+        for task_id in task_ids:
+            task = Task.query.get(task_id)
+            task.goal = goal
+        db.session.commit()
+        response_task_ids = []
+        for task in goal.tasks:
+            response_task_ids.append(task.task_id)
+        new_tasks_response = {
+            "id": goal.goal_id,
+            "task_ids": response_task_ids
+        }
+        return jsonify(new_tasks_response), 200
+        
+    elif request.method == "GET":
```

Comment:
👍 Good work! Consider refactoring with a dictionary helper method in `Task` and `Goal`.